OpenRA.Mods.Common/Traits/Health.cs

```diff
@@ -95,11 +120,11 @@ public void Resurrect(Actor self, Actor repairer)
 				PreviousDamageState = DamageState.Dead,
 			};
 
-			foreach (var nd in self.TraitsImplementing<INotifyDamage>()
-					.Concat(self.Owner.PlayerActor.TraitsImplementing<INotifyDamage>()))
+			foreach (var nd in notifyDamage
```

Comment:
We can avoid allocating an enumerator here by adding a second `foreach` loop for the player traits.

Reply:
Done. Assuming it would have the same effect there, I've done the same for the uncacheable AppliedDamage loops.